### ISSUE TITLE
distrib listenUNIX should use the usePID=1 lock system

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2295]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2295|
|Type|defect|
|Created|2006-12-13 04:43:36Z|
|Branch|https://github.com/twisted/twisted/tree/distrib-wantpid-2295-2|

Without the lock system, a crashed distrib web server can't restart automatically, cause the unix socket is left lying around.

<details><summary>Searchable metadata</summary>

```
trac-id__2295 2295
type__defect defect
reporter__itamarst itamarst
priority__normal normal
milestone__ 
branch__branches_distrib_wantpid_2295_2 branches/distrib-wantpid-2295-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1165985016000000 1165985016000000
changetime__1224445324000000 1224445324000000
version__None None
owner__exarkun exarkun
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Instead of making distrib pass `wantPID=True` unconditionally, I made `--personal` start a server listening on the value passed to `--port`.

I also extended strports UNIX support to accept a `wantPID=1` parameter.

The default behavior is the same, but if a user wants pidfile-based locking on their distrib servers, they can get it.

Build results: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/distrib-wantpid-2295

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix commented|
|-|-|

wantPID is a bad name. It's too bad that it's already used in listenUNIX, but if we're going to expose it to strports we should use a better name.

writePIDFile sounds like a better option.

I also wonder why we don't just default to writing PID files.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Cool, please merge. Changing the name is an unrelated issue, open another ticket if you want.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix commented|
|-|-|

I don't have much of an excuse, since my comment above was not explicitly a review, but I'm curious to know whether the points I raised were dealt with in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

They weren't at all.  I'll think about them before merging, even though therve said I could, and maybe address them and re-submit for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

I renamed the parameter in strports (not elsewhere) and change its default to true.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_8"></a>|@ralphm set owner to @exarkun|
|-|-|

Looks good to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [24989]) Branching to 'distrib-wantpid-2295-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [24991]) Merge distrib-wantpid-2295-2

Author: exarkun
Reviewer: ralphm
Fixes: [#2295](https://github.com/twisted/twisted/issues/2295)

Add an optional parameter to the AF_FILE style of strport
description, `lockfile`.  By default, a "unix:" strport
will attempt to acquire a lock associate with the socket
specified.  If `lockfile=0` is given, this behavior will
be disabled.

`twisted.web.distrib` is also changed to allow a strports
description for the port it listens on, allowing a lock
file to be used or not with the port it listens on.
